### PR TITLE
Dockerfile fixes

### PR DESCRIPTION
**Pull request by arangatang**
*Issue #, if available:*
solves #945

*Description of changes:*
the mxnet dependencies are now installed.
also fixed indentation of the Dockerfile.gpu. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,7 @@ FROM python:3.7
 
 ADD . /gluonts
 
+RUN pip install mxnet==1.6
```
Comment (jaheba) [inline Dockerfile:5]:
```suggestion
RUN pip install -r /gluonts/requirements/requirements-mxnet.txt
```

This should ensure that we install the currently recommended mxnet version.

File: Dockerfile.gpu [hunk 2]
```diff
@@ -14,6 +14,7 @@ RUN echo '#!/bin/bash\necho Building container with GPU support' > /usr/bin/nvid
 
 ADD . /gluonts
 
+RUN pip3.7 install mxnet-cu92
```
Comment (AaronSpieler) [inline Dockerfile.gpu:17]: This also be `mxnet-cu92==1.6` or whatever version is specified in the requirements.

Comment (arangatang) [reply Dockerfile.gpu:17]: in the requirements the version is defined as >=1.6.0 so ill use this instead

Comment (AaronSpieler) [reply Dockerfile.gpu]: ok, but then why not `mxnet-cu92>=1.6`

**Comment by AaronSpieler** [comment]
- I added support for R in the cpu container.
    - Tested this using the `ets` method and the new experimentation framework.
- The resulting image is about 30% larger and takes much longer to build.

I like this because the cpu image now intuitively supports all our Predictors.
We could think about having a separate cpu Dockerfile with R support though.

What do you think @jaheba, @arangatang

**Comment by jaheba** [comment]
> We could think about having a separate cpu Dockerfile with R support though.

I think I prefer that approach.

Maybe we can generally extend the use of https://github.com/awslabs/gluon-ts/tree/master/examples/dockerfiles

**Comment by AaronSpieler** [comment]
> 
> 
> > We could think about having a separate cpu Dockerfile with R support though.
> 
> I think I prefer that approach.
> 
> Maybe we can generally extend the use of https://github.com/awslabs/gluon-ts/tree/master/examples/dockerfiles

Great idea. We should really move it to that folder.
Did that Dockerfile.r work? @Schmedu I tried installing the packages that way and it when I tested the shell it didn't work.

**Comment by Schmedu** [comment]
@AaronSpieler yes, it works. I just checked it.

**Comment by jaheba** [comment]
How about we move the Dockerfiles to `examples/dockerfiles` but we link to it in the readme?

**Comment by arangatang** [comment]
Grouping them together would make sense, i agree.

File: examples/dockerfiles/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.7
+
+RUN apt-get update && apt-get install -y r-base
+RUN R -e "install.packages('forecast',dependencies=TRUE, repos='http://cran.rstudio.com/')"
+RUN R -e "install.packages('nnfor',dependencies=TRUE, repos='http://cran.rstudio.com/')"
```
[the remark this answers is not in the document]
Comment (arangatang) [reply examples/dockerfiles/Dockerfile:5]: Yeah, just saw that now too. I think ill remove this and add a readme

File: examples/dockerfiles/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Instructions for using the Dockerfiles
+
+To build these dockerfiles go to the Gluon-TS root directory and run:
```
Comment (jaheba) [inline examples/dockerfiles/README.md:3]: I think what also works is doing

    docker build ../.. -f <dockerfile>

in the current folder. Maybe we should list both ways.

File: examples/dockerfiles/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+```bash 
+docker build . -f examples/dockerfiles/<the dockerfile>
+```
+
+To run the built image run:
```
Comment (jaheba) [inline examples/dockerfiles/README.md:9]: "run ... run" is a bit confusing.

File: examples/dockerfiles/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+```bash
+docker run <image_id> <shell_params>
```
Comment (jaheba) [inline examples/dockerfiles/README.md:12]: Let's maybe leave this out, and just say that these images can be used on SageMaker?

File: examples/dockerfiles/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+For more information about the shell and the available params, see the [shell documentation](https://github.com/awslabs/gluon-ts/tree/master/src/gluonts/shell) or run:
+
+```bash
+docker run <image_id> --help
+```
+
+## How to choose between the dockerfiles
+
+* Dockerfile
+    - This is for running Gluon-TS algorithms on a machine using only the CPU. This image is the most basic one and allows the use of most of the gluonts forecasters.
```
Comment (jaheba) [inline examples/dockerfiles/README.md:24]: I think we can abbreviate this by quite a bit, e.g.

```suggestion
    - Gluon-TS models using CPU.
```

Comment (arangatang) [reply examples/dockerfiles/README.md]: Oki

File: examples/dockerfiles/README.md (hunk 6 above)
Comment (arangatang) [reply examples/dockerfiles/README.md]: Removed this

File: examples/dockerfiles/README.md (hunk 5 above)
Comment (arangatang) [reply examples/dockerfiles/README.md]: Removed this

File: examples/dockerfiles/README.md (hunk 4 above)
Comment (arangatang) [reply examples/dockerfiles/README.md]: Added this

File: examples/dockerfiles/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+# Instructions for using the Dockerfiles
+
+To build these dockerfiles go to the Gluon-TS root directory and enter this command:
+
+```bash 
+docker build . -f examples/dockerfiles/<the dockerfile>
```
Comment (jaheba) [inline examples/dockerfiles/README.md:6]:
```suggestion
docker build . -f examples/dockerfiles/<dockerfile>
```

File: examples/dockerfiles/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+```bash
+docker build ../.. -f <dockerfile>
+```
+
+The built images are compatible with sagemaker.
+For more information about the shell and the available params, see the [shell documentation](https://github.com/awslabs/gluon-ts/tree/master/src/gluonts/shell) or run:
+
+```bash
+docker run <image_id> --help
+```
```
Comment (jaheba) [inline examples/dockerfiles/README.md:20]: Let's maybe just remove this as well.

```suggestion
```

File: examples/dockerfiles/README.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+## How to choose between the dockerfiles
+
+* Dockerfile
+    - Gluon-TS models using CPU.
+* Dockerfile.gpu
+    -  Gluon-TS models on a GPU accelerated machine. 
+* Dockerfile.r
+    - This is used when one wants to run the models in the [gluonts.model.r_forecast](https://gluon-ts.mxnet.io/api/gluonts/gluonts.model.r_forecast.html) package.
```
Comment (jaheba) [inline examples/dockerfiles/README.md:29]:
```suggestion
    - This provides dependencies for models defined in [gluonts.model.r_forecast](https://gluon-ts.mxnet.io/api/gluonts/gluonts.model.r_forecast.html).
```

**Pull request merged**